pkg/framework/aries/default.go

```diff
@@ -122,11 +150,102 @@ func setAdditionalDefaultOpts(frameworkOpts *Aries) error {
 		frameworkOpts.msgSvcProvider = &noOpMessageServiceProvider{}
 	}
 
-	// TODO add SecretLock creation here.. #1150
+	return nil
+}
+
+func createMasterKeyAndSecretLock(opts *Aries) error {
+	keySize := sha256.Size
+
+	passphrase, salt, err := getPassphraseAndSalt(opts, keySize)
+	if err != nil {
+		return err
+	}
+
+	masterLocker, err := hkdf.NewMasterLock(passphrase, sha256.New, salt)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(defMKPath, os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+
+	fileName := file.Name()
+
+	fStat, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	if fStat.Size() < int64(base64.URLEncoding.EncodedLen(keySize)) {
+		// if file content doesn't contain a valid key size, generate new key, encrypt and overwrite file
+		err = generateAndWriteMK(file, keySize, masterLocker)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = file.Close()
+	if err != nil {
+		return err
+	}
+
+	r, err := local.MasterKeyFromPath(fileName)
+	if err != nil {
+		return err
+	}
+
+	s, err := local.NewService(r, masterLocker)
+	if err != nil {
+		return err
+	}
+
+	opts.secretLock = s
+
+	return nil
+}
+
+func generateAndWriteMK(file io.Writer, keySize int, masterLocker secretlock.Service) error {
+	masterKeyContent := random.GetRandomBytes(uint32(keySize))
+
+	masterLockEnc, err := masterLocker.Encrypt("", &secretlock.EncryptRequest{
+		Plaintext: string(masterKeyContent)})
+	if err != nil {
+		return err
+	}
+
+	_, err = file.Write([]byte(masterLockEnc.Ciphertext))
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+func getPassphraseAndSalt(opts *Aries, keySize int) (string, []byte, error) {
+	passphrase := "secretPwd"
```

Comment:
Shouldn't this input be required from the user?

Reply:
to ease the creation of the Aries framework and not impact existing code (namely BDD tests) that don't use the new KMS (and SecretLock), the framework will use this default passphrase in case the passphrase option (and KMS/SecretLock options) are not passed

Reply:
to clarify, this is supposed to be temporary until we flip legacyKMS into KMS, then either passphrase or SecretLock options may become mandatory .. it's to be reviewed in the future

Reply:
Right, so what should be the default secretlock implementation?

Reply:
there is no more passphrase being passed as an option.

the default behaviour (if no SecretLock option is passed to `New()`) is to create a SecretLock with a master key file generated and stored `unprotected`  in a file. The default service will use the master key as is (without encryption/decryption of master key) to wrap KMS keys.

For a more secure approach, one must create a SecretLock option with secret expansion to protect the master key prior to storing in a path. Examples are in unit tests and a README document will be pushed soon to explain how to do this.